File: app/src/main/java/com/megablok10/app/breach/Tier.kt
package com.megablok10.app.breach

/**
 * Три уровня сложности — общие для контейнера, демона и лут-слота. Один enum,
 * а не отдельные Int/String на каждой сущности: "демон тира HARD извлекает
 * слот тира HARD и ниже" сравнивается напрямую через compareTo по level, без
 * ручной синхронизации трёх разных представлений одного и того же понятия.
 */
enum class Tier(val level: Int, val label: String) {
    BASE(1, "База"),
    HARD(2, "Сложно"),
    NIGHTMARE(3, "Кошмар");

    /** Тир a "покрывает" тир b, если a не ниже b — демон высокого тира извлекает и более простые слоты. */
    fun covers(other: Tier): Boolean = level >= other.level

    companion object {
        fun fromLevel(level: Int): Tier = entries.firstOrNull { it.level == level } ?: BASE
    }
}

File: admin-web/server/src/lib/tier.ts
/** Зеркало enum class Tier в app/src/main/java/com/megablok10/app/breach/Tier.kt — держать в синхроне вручную. */
export const TIER_LEVEL: Record<string, number> = { BASE: 1, HARD: 2, NIGHTMARE: 3 };

export function tierLevel(tierName: string): number {
  return TIER_LEVEL[tierName] ?? 1;
}
